include/kernels/covDev.cuh

```diff
@@ -31,6 +31,7 @@ namespace quda
 
     F out;      /** output vector field */
     const F in; /** input vector field */
+    const F in_pack; /** input vector field used in packing to be able to independently resetGhost */
```

Comment:
Why is this necessary/is there a cleaner way to do this? I see `in_pack` is always initialized to `in`, so I'm not so worried about memory overhead bloat, but it seems a little hacky.

Reply:
In `dslash.h` around line 140 
we may need different ghosts for packing 
```cpp
      arg.in.resetGhost(in, ghost);

      if (arg.pack_threads && arg.kernel_type == INTERIOR_KERNEL) {
        arg.blocks_per_dir = tp.aux.x;
        arg.setPack(true, this->packBuffer); // need to recompute for updated block_per_dir
        arg.in_pack.resetGhost(in, this->packBuffer);
```

Reply:
Sure, my thought is related to including different/multiple ghosts within a single `ColorSpinorField` as opposed to using two separate `ColorSpinorField`s with different ghosts.

Reply:
I am inclined to put this to a fixed later and file an issue for looking into this later. But will think about it more.

Reply:
Yes, I'd be happy deferring this.  It could be reexamined circa multi-RHS.

Reply:
#1117